src/Filter/NestedFilter.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace EasyCorp\Bundle\EasyAdminBundle\Filter;
+
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ObjectManager;
+use EasyCorp\Bundle\EasyAdminBundle\Contracts\Filter\FilterInterface;
+use EasyCorp\Bundle\EasyAdminBundle\Dto\EntityDto;
+use EasyCorp\Bundle\EasyAdminBundle\Dto\FieldDto;
+use EasyCorp\Bundle\EasyAdminBundle\Dto\FilterDataDto;
+
+/**
+ * @author Brandon Marcachi <brandon.marcachi@gmail.com>
+ */
+final class NestedFilter implements FilterInterface
+{
+    use FilterTrait;
+
+    public const FORM_OPTION_WRAPPED_FILTER = 'attr.wrapped_filter';
+
+    public const PATH_SEPARATOR_EXPECTED = '.';
+    public const PATH_SEPARATOR = '_';
+
+    /** @var FilterInterface */
+    private $wrappedFilter;
+
+    public static function new(string $propertyName, string $label = null): self
+    {
+        throw new \RuntimeException('Instead of this method, use the "wrap()" method.');
+    }
+
+    public static function wrap(FilterInterface $filter): FilterInterface
+    {
+        $filterDto = $filter->getAsDto();
+        $property = $filterDto->getProperty();
+
+        if (false === strpos($property, self::PATH_SEPARATOR_EXPECTED)) {
+            return $filter;
+        }
+
+        return (new self())
+            ->setFilterFqcn(__CLASS__)
+            ->setProperty($property)
+            ->setFormType($filterDto->getFormType())
+            ->setFormTypeOptions($filterDto->getFormTypeOptions())
+            ->setWrappedFilter($filter)
+        ;
+    }
+
+    public function apply(QueryBuilder $queryBuilder, FilterDataDto $filterDataDto, ?FieldDto $fieldDto, EntityDto $entityDto): void
+    {
+        $propertyPath = $filterDataDto->getProperty();
+
+        [$targetClassMetadata, $targetProperty] = self::extractTargets(
+            $queryBuilder->getEntityManager(),
+            $entityDto->getFqcn(),
+            $propertyPath
+        );
+
+        $wrappedEntityDto = new EntityDto($targetClassMetadata->getName(), $targetClassMetadata);
+        $wrappedFilter = $this->getWrappedFilter();
+        $wrappedFilterDto = $wrappedFilter->getAsDto();
+        $wrappedFilterDto->setProperty($targetProperty);
+
+        // Apply required left joins and get the alias we have to work with
+        $alias = $this->applyLeftJoins($queryBuilder, $filterDataDto->getEntityAlias(), $propertyPath);
```

Comment:
This adds support for embeddables.

```suggestion
        if ($entityDto->getFqcn() === $targetClassMetadata->getName()) {
            $alias = $filterDataDto->getEntityAlias();
        } else {
            // Apply required left joins and get the alias we have to work with
            $alias = $this->applyLeftJoins($queryBuilder, $filterDataDto->getEntityAlias(), $propertyPath);
        }
```